lib/aytests/image_builder.rb

```diff
@@ -57,6 +57,33 @@ def initialize(sources_dir: nil, work_dir: nil, files_dir: nil, provider: :libvi
       @provider = provider
     end
 
+    # Due an aborted run there could be old stuff which should be
+    # removed before starting a new run.
+    #
+    def cleanup_environment
+      if provider == :libvirt
+        pool_lines = Cheetah.run(["sudo", "virsh", "pool-list"], stdout: :capture).lines.drop(2)
+        pool = pool_lines.collect { |line| line.split.first }.compact
```

Comment:
When using the `{}` notation for blocks, it's preferred to use a single-letter variable name:

```
pool = pool_lines.collect { |l| l.split.first }.compact
```